protocol/contracts/libraries/Minting/LibWellMinting.sol

```diff
@@ -183,6 +184,56 @@ library LibWellMinting {
         }
     }
 
+    function cappedReservesDeltaB(address well) internal view returns (int256) {
+        if (well == C.BEAN) {
+            return 0;
+        }
+
+        // get first pump from well
+        Call[] memory pumps = IWell(well).pumps();
+        address pump = pumps[0].target;
+
+        // well address , data[]
+        uint256[] memory instReserves = ICappedReservesPump(pump).readCappedReserves(
+            well,
+            pumps[0].data
+        );
+        // Get well tokens
+        IERC20[] memory tokens = IWell(well).tokens();
+
+        // Get ratios and bean index
+        (uint256[] memory ratios, uint256 beanIndex, bool success) = LibWell.getRatiosAndBeanIndex(
+            tokens
+        );
+
+        // HANDLE FAILURE
+        // If the Bean reserve is less than the minimum, the minting oracle should be considered off.
+        if (instReserves[beanIndex] < C.WELL_MINIMUM_BEAN_BALANCE) {
+            return 0;
+        }
+
+        // If the USD Oracle oracle call fails, the minting oracle should be considered off.
+        if (!success) {
+            return 0;
+        }
+
+        // Get well function
+        Call memory wellFunction = IWell(well).wellFunction();
+
+        // Delta B is the difference between the target Bean reserve at the peg price
+        // and the instantaneous Bean balance in the Well.
+        int256 deltaB = int256(
+            IBeanstalkWellFunction(wellFunction.target).calcReserveAtRatioSwap(
+                instReserves,
+                beanIndex,
+                ratios,
+                wellFunction.data
+            )
+        ).sub(int256(instReserves[beanIndex]));
+
+        return deltaB;
+    }
+
     // Remove in next BIP.
     function checkShouldTurnOnMinting(address well) internal view returns (bool) {
```

Comment:
can we remove this

Reply:
Yes, removed here: https://github.com/BeanstalkFarms/Beanstalk/commit/7922b244d25ace3d3eb779bc1187cc437f0e23c2